Yosemite/Yosemite/Stores/AppSettingsStore.swift

```diff
@@ -467,6 +485,62 @@ private extension AppSettingsStore {
     }
 }
 
+// MARK: - Products Settings
+//
+private extension AppSettingsStore {
+    func loadProductsSettings(siteID: Int64, onCompletion: (Result<StoredProductSettings, Error>) -> Void) {
+        guard let allSavedSettings: [StoredProductSettings] = try? fileStorage.data(for: productsSettingsURL) else {
+            let error = AppSettingsStoreErrors.noProductsSettings
+            onCompletion(.failure(error))
+            return
+        }
+
+        let settings = allSavedSettings.filter { $0.siteID == siteID }
+
+        guard let settingsUnwrapped = settings.first else {
+            let error = AppSettingsStoreErrors.noProductsSettings
+            onCompletion(.failure(error))
```

Comment:
I just noticed this. I'm not sure how I feel about returning with an error because I believe it's not really an error but just a “there is no data“ state (`null`). And that is expected. But based on how we use this `Action`, I guess the difference is not a big deal. 😄